From 33b3b55abb104b3817072f5b95bdfec459f96633 Mon Sep 17 00:00:00 2001
From: bohu <bohu@google.com>
Date: Fri, 15 Dec 2017 11:20:58 -0800
Subject: [PATCH 1/5] mediacodec: add seccomp policy for x86 and x86_64

We had it for arm but not for x86; this cl adds it.

BUG: 70507558

Test: run cts -m CtsMediaTestCases and mediacodec should not
crash due to minijail
Test: launch emulator and mediacodec should not be killed by signal 31
Test: run cts -m CtsSecurityHostTestCases
-t android.security.cts.ProcessMustUseSeccompTest#testOmxHalHasSeccompFilter
should pass
Test: run cts -m CtsSecurityTestCases -t
android.security.cts.StagefrightTest#testStagefright_cve_2017_0600
Test: run cts -m CtsSecurityTestCases -t
android.security.cts.StagefrightTest#testStagefright_cve_2016_0842

Note: I only tested on emulator, not on real x86 device

Change-Id: I63bebc17eb2ae29df71823f8277f058ca7fd9625
Merged-In: I63bebc17eb2ae29df71823f8277f058ca7fd9625
---
 services/mediacodec/Android.mk                     |  3 +-
 .../seccomp_policy/mediacodec-x86.policy           | 69 ++++++++++++++++++++++
 2 files changed, 71 insertions(+), 1 deletion(-)
 create mode 100644 services/mediacodec/seccomp_policy/mediacodec-x86.policy

diff --git a/services/mediacodec/Android.mk b/services/mediacodec/Android.mk
index faeb0a7..ba2ab22 100644
--- a/services/mediacodec/Android.mk
+++ b/services/mediacodec/Android.mk
@@ -19,6 +19,7 @@ include $(BUILD_SHARED_LIBRARY)
 # service executable
 include $(CLEAR_VARS)
 LOCAL_REQUIRED_MODULES_arm := mediacodec.policy
+LOCAL_REQUIRED_MODULES_x86 := mediacodec.policy
 LOCAL_SRC_FILES := main_codecservice.cpp
 LOCAL_SHARED_LIBRARIES := \
     libmedia_omx \
@@ -45,7 +46,7 @@ LOCAL_INIT_RC := android.hardware.media.omx@1.0-service.rc
 include $(BUILD_EXECUTABLE)
 
 # service seccomp policy
-ifeq ($(TARGET_ARCH), $(filter $(TARGET_ARCH), arm arm64))
+ifeq ($(TARGET_ARCH), $(filter $(TARGET_ARCH), x86 x86_64 arm arm64))
 include $(CLEAR_VARS)
 LOCAL_MODULE := mediacodec.policy
 LOCAL_MODULE_CLASS := ETC
diff --git a/services/mediacodec/seccomp_policy/mediacodec-x86.policy b/services/mediacodec/seccomp_policy/mediacodec-x86.policy
new file mode 100644
index 0000000..dc2c04f
--- /dev/null
+++ b/services/mediacodec/seccomp_policy/mediacodec-x86.policy
@@ -0,0 +1,69 @@
+# Copyright (C) 2017 The Android Open Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+read: 1
+mprotect: 1
+prctl: 1
+openat: 1
+getuid32: 1
+writev: 1
+ioctl: 1
+close: 1
+mmap2: 1
+fstat64: 1
+madvise: 1
+fstatat64: 1
+futex: 1
+munmap: 1
+faccessat: 1
+_llseek: 1
+lseek: 1
+clone: 1
+sigaltstack: 1
+setpriority: 1
+restart_syscall: 1
+exit: 1
+exit_group: 1
+rt_sigreturn: 1
+ugetrlimit: 1
+readlinkat: 1
+_llseek: 1
+fstatfs64: 1
+pread64: 1
+mremap: 1
+dup: 1
+set_tid_address: 1
+write: 1
+nanosleep: 1
+
+# for attaching to debuggerd on process crash
+socketcall: 1
+sigaction: 1
+tgkill: 1
+rt_sigprocmask: 1
+fcntl64: 1
+rt_tgsigqueueinfo: 1
+geteuid32: 1
+getgid32: 1
+getegid32: 1
+getgroups32: 1
+getdents64: 1
+pipe2: 1
+ppoll: 1
+
+# Required by AddressSanitizer
+gettid: 1
+sched_yield: 1
+getpid: 1
+gettid: 1
-- 
1.9.1

